docker.go

```diff
@@ -829,6 +830,22 @@ func (p *DockerProvider) BuildImage(ctx context.Context, img ImageBuildInfo) (st
 	return repoTag, nil
 }
 
+func waitContainerReadyWithLog(ctx context.Context, container *DockerContainer) error {
```

Comment:
Why the `withLog` suffix to the name? Is it because it prints out the "Waiting for container..."??

Please also see my other comment about the need to wait twice, which could eliminate the need to extract the method into a reusable function. wdyt?

Reply:
> Why the withLog suffix to the name? Is it because it prints out the "Waiting for container..."??
Yes, I wanted to avoid duplicating the log message and put it in a separate function. If you would like another function name, feel free to ask.

Reply:
Ok, I think I now understand. Could we make the function `waitForContainerReady`? The withLog suffix could be an implementation detail, and there is no other alternative, so I think it's fine to not add it to the function name.